gneiss/sort.py

```diff
@@ -114,3 +114,119 @@ def niche_sort(table, gradient, niche_estimator=mean_niche_estimator):
     table = table.reindex(index=gradient.index,
                           columns=est_niche.index)
     return table
+
+
+def ladderize(tree, ascending=True):
+    """
+    Sorts tree according to the size of the subtrees.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+       Input tree where leafs correspond to features
+       contained in the index in `grouping1.
+    gradient : pd.Series
+       Gradient where the index correspond to features.
+
+    Returns
+    -------
+    skbio.TreeNode
+       A tree whose tips are sorted along the gradient.
+
+    Examples
+    --------
+    >>> from skbio import TreeNode
+    >>> tree = TreeNode.read([u'((a,b)c, ((g,h)e,f)d)r;'])
+    >>> print(tree.ascii_art())
+                        /-a
+              /c-------|
+             |          \-b
+    -r-------|
+             |                    /-g
+             |          /e-------|
+              \d-------|          \-h
+                       |
+                        \-f
+    >>> sorted_tree = ladderize(tree)
+    >>> print(sorted_tree.ascii_art())
+                        /-a
+              /c-------|
+             |          \-b
+    -r-------|
+             |          /-f
+              \d-------|
+                       |          /-g
+                        \e-------|
+                                  \-h
+    """
+    sorted_tree = tree.copy()
+    # Note that this is an extremely slow
+    # See https://github.com/biocore/gneiss/issues/58
+    for n in sorted_tree.postorder(include_self=True):
+        sizes = [len(list(k.tips())) for k in n.children]
+        if ascending:
+            idx = np.argsort(sizes)
+        else:
+            idx = np.argsort(sizes)[::-1]
+        n.children = [n.children[i] for i in idx]
+    return sorted_tree
+
+
+def order_tips(tree, gradient, ascending=True):
```

Comment:
Yes!  I agree.  But its not clear to me how to best do this.  I've raised an issue in #60 to further discuss.  But I'd like to get this in first, since it will significantly clean up the notebooks / analyses.